### PR TITLE
Package mazeppa.0.4.0

### PR DESCRIPTION
**Pull request by hirrolot**
### `mazeppa.0.4.0`
A modern supercompiler for call-by-value functional languages

---
* Homepage: https://github.com/mazeppa-dev/mazeppa
* Source repo: git+https://github.com/mazeppa-dev/mazeppa.git
* Bug tracker: https://github.com/mazeppa-dev/mazeppa/issues

---
:camel: Pull-request generated by opam-publish v2.3.0

### CODE REVIEW

File: packages/mazeppa/mazeppa.0.4.0/opam [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  "ocaml" {>= "4.14"}
+  "dune" {>= "3.14"}
+  "pprint"
+  "checked_oint" {>= "0.2.1"}
+  "ppx_deriving"
+  "ppx_string_interpolation"
+  "ppx_yojson_conv"
+  "ppx_blob" {>= "0.9.0"}
+  "bisect_ppx"
+  "menhir"
```
Comment (shonfeder) [inline packages/mazeppa/mazeppa.0.4.0/opam:20]: Looks like we are hitting a lower bounds error due to this unconstrained dependency on menhir:

```
#=== ERROR while compiling mazeppa.0.4.0 ======================================#
# context              2.4.0~alpha1~dev | linux/x86_64 | ocaml-base-compiler.4.14.2 | pinned(https://github.com/mazeppa-dev/mazeppa/archive/refs/tags/v0.4.0.tar.gz)
# path                 ~/.opam/4.14/.opam-switch/build/mazeppa.0.4.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p mazeppa -j 39 @install
# exit-code            1
# env-file             ~/.opam/log/mazeppa-7-19e53b.env
# output-file          ~/.opam/log/mazeppa-7-19e53b.out
### output ###
# File "lib/dune", line 22, characters 0-26:
# 22 | (menhir
# 23 |  (modules parser))
# (cd _build/.sandbox/49aaa9762fd354fd4b60213f6d1a699d/default && /home/opam/.opam/4.14/bin/menhir lib/parser.mly --base lib/parser --infer-write-query lib/parser__mock.ml.mock)
# /home/opam/.opam/4.14/bin/menhir: unknown option '--infer-write-query'.
```

It looks like this flag was added in version [20180523](https://gitlab.inria.fr/fpottier/menhir/-/blob/master/CHANGES.md?ref_type=heads#20180523)

Let's try

```suggestion
  "menhir" {>= "20180523"}
```

File: packages/mazeppa/mazeppa.0.4.0/opam [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  "ocaml" {>= "4.14"}
+  "dune" {>= "3.14"}
+  "pprint"
+  "checked_oint" {>= "0.2.1"}
+  "ppx_deriving"
+  "ppx_string_interpolation"
+  "ppx_yojson_conv"
+  "ppx_blob" {>= "0.9.0"}
+  "bisect_ppx"
+  "menhir" {>= "20180523"}
```
Comment (shonfeder) [inline packages/mazeppa/mazeppa.0.4.0/opam:20]: Ah, and we get one more lower bounds error now

```
#=== ERROR while compiling mazeppa.0.4.0 ======================================#
# context              2.4.0~alpha1~dev | linux/x86_64 | ocaml-base-compiler.4.14.2 | pinned(https://github.com/mazeppa-dev/mazeppa/archive/refs/tags/v0.4.0.tar.gz)
# path                 ~/.opam/4.14/.opam-switch/build/mazeppa.0.4.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p mazeppa -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/mazeppa-7-9f5ad1.env
# output-file          ~/.opam/log/mazeppa-7-9f5ad1.out
### output ###
# File "lib/dune", line 22, characters 0-26:
# 22 | (menhir
# 23 |  (modules parser))
# Error: Rule failed to generate the following targets:
# - lib/parser.conflicts
```

I think indicating that we need at least [20180905](https://gitlab.inria.fr/fpottier/menhir/-/blob/master/CHANGES.md?ref_type=heads#20180905):

```suggestion
  "menhir" {>= "20180905"}
```

**Comment by shonfeder** [comment]
> My concern with the first approach is that version requirements in a GitHub release and Opam become inconsistent

Yeah, I've also had that struggle! From the opam repo's perspective, the source of truth for the package data is the opam file checked into the repo here, and the opam file in our package sources is just some redundant thing ignored during installation. But from our perspective as package authors, we may want to keep these two configurations in perfect sync.

It is common practice to fix the opam file when making PRs into this repo and just release it. But if you prefer to update the configuration in the source and then reflect that update here with a new version, that is also fine. The latter is just a bit more work.

Would you like me to merge this in as is, or do you want to cut a new release?

**Comment by hirrolot** [comment]
Thanks for a detailed explanation! Let me apply your fix to my repository and make a new PR with an updated patch version. I'll close this PR and open another one soon.

**Pull request closed without merging**